lucidity/rpc/rpc.go

```diff
@@ -99,37 +87,25 @@ func (s *server) Update(ctx context.Context, req *pb.UpdateRequest) (*pb.UpdateR
 	if !ok || v.(*pb.UpdateRequest).Version != req.Version {
 		s.recalculateValidVersions()
 	}
-	validVersion := s.IsValidVersion(req.Version)
-	if !validVersion {
-		req.Healthy = false
-		req.Status = "Invalid version"
-	}
-	s.updateMaps(req)
-	return &pb.UpdateResponse{ShouldDisable: req.Disabled || !validVersion}, nil
+	s.checkWorkerHealth(req)
+	return &pb.UpdateResponse{ShouldDisable: req.Disabled || !req.Healthy}, nil
 }
 
-func (s *server) updateMap(m map[string]struct{}, key string, in bool) float64 {
-	if in {
-		m[key] = struct{}{}
-	} else {
-		delete(m, key)
+func (s *server) checkWorkerHealth(req *pb.UpdateRequest) {
```

Comment:
Then perhaps we could remove the `if req.Healthy` from the if check below, so that it updates the status to "Too long since last update" even if there's something else wrong with the worker

Reply:
Hmm, yeah good point. The worker might complain about something else, but I guess we dont care if it's not up-to-date.